public/app/features/serviceaccounts/components/ServiceAccountProfile.tsx

```diff
@@ -53,7 +53,7 @@ export function ServiceAccountProfile({ serviceAccount, timeZone, onChange }: Pr
           <ServiceAccountProfileRow
             label="Name"
             value={serviceAccount.name}
-            onChange={onNameChange}
+            onChange={!serviceAccount.isExternal ? onNameChange : undefined}
```

Comment:
Is that a common pattern in Typescript? 
@Clarity-89 for insights

Reply:
Yeah, this is fine.